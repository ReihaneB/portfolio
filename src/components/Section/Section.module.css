.root {
  margin-top: 150px;
  margin-bottom: 150px;

  @media (width <= 768px) {
    margin-top: 100px;
    margin-bottom: 100px;
  }
}
