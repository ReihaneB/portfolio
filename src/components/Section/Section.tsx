import type { SectionProps } from './Section.d';
import styles from './Section.module.css';

function Section({
  children,
  id,
  ...rest
}: SectionProps) {
  return (
    <section
      className={styles.root}
      id={id}
      {...rest}
    >
      {children}
    </section>
  );
}

export default Section;
