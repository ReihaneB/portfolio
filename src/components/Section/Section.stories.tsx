import type { Meta, StoryObj } from '@storybook/react';
import { expect, userEvent, waitFor, within } from '@storybook/test';

import { lokiPlayDecorator } from '@/testUtils/loki/play-decorator';

import Section from './Section';

const meta: Meta<typeof Section> = {
  title: 'Components/Section',
  component: Section,
};

export default meta;
type Story = StoryObj<typeof Section>;

const defaultArgs = {
  children: 'Section content',
  id: 'section-id',
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const NotFocusTab: Story = {
  args: {
    ...defaultArgs,
  },
  play: lokiPlayDecorator(
    async ({ canvasElement }) => {
      const canvas = within(canvasElement);
      const section = canvas.getByText('Section content');

      expect(section).toBeInTheDocument();

      await userEvent.tab();

      await waitFor(() => expect(section).not.toHaveFocus());
    }
  ),
};

export const SemanticSection: Story = {
  args: {
    ...defaultArgs,
  },
  name: 'Should render a semantic section element',
  play: lokiPlayDecorator(
    async ({ canvasElement }) => {
      const canvas = within(canvasElement);

      const section = canvas.getByText('Section content');

      expect(section).toBeInTheDocument();
      expect(section.tagName).toBe('SECTION');
    }
  ),
};

