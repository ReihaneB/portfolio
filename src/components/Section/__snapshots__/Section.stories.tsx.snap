// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Section NotFocusTab play-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <section class="src-components-Section-Section-module__root--wtqbG"
           id="section-id"
  >
    Section content
  </section>
</main>
`;

exports[`Components/Section Playground smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <section class="src-components-Section-Section-module__root--wtqbG"
           id="section-id"
  >
    Section content
  </section>
</main>
`;

exports[`Components/Section SemanticSection play-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <section class="src-components-Section-Section-module__root--wtqbG"
           id="section-id"
  >
    Section content
  </section>
</main>
`;
